hadoop-common-project/hadoop-nfs/src/main/java/org/apache/hadoop/oncrpc/RpcUtil.java

```diff
@@ -170,15 +172,18 @@ public void channelRead(ChannelHandlerContext ctx, Object msg)
    */
   @ChannelHandler.Sharable
   private static final class RpcUdpResponseStage extends
-      ChannelInboundHandlerAdapter {
+          SimpleChannelInboundHandler<RpcResponse> {
+    public RpcUdpResponseStage() {
+      // do not auto release the RpcResponse message.
```

Comment:
Should we release `response` in `channelRead0`?
```java
      ctx.writeAndFlush(new DatagramPacket(buf,
              (InetSocketAddress) response.recipient()))
          .addListener(future -> response.release());
```

Reply:
no... i checked with io.netty.leakDetection.level=PARANOID, and it would actually throw an exception because the refcount would go negative.
The response is released. Looks like IdleStateHandler releases the object for us.

Reply:
Thanks for checking it